channeldb/payments.go

```diff
@@ -109,6 +109,10 @@ const (
 	// balance to complete the payment.
 	FailureReasonInsufficientBalance FailureReason = 4
 
+	// FailureReasonMPPTimeout indicates that we attempted a mpp payment which
+	// timed out.
+	FailureReasonMPPTimeout FailureReason = 5
```

Comment:
So these are payment-level failures. I don't think that mpp timeout is a valid final payment-level failure, because we just need to try again (eventually). I would for now just map that onto `FailureReasonError`.

Reply:
Seems like a slightly strange mapping to me because `FailureReasonError` in channelDB is for unexpected errors (according to its comment at least). But it's good enough since we don't expect MPP timeouts.

Reply:
That is indeed slightly strange. We currently mix up unexpected (internal) errors and certain received failure messages (search for `reasonError` in `result_interpretation.go`). Needs to be fixed at some point.